pkg/util/httprequest/httprequest_test.go

```diff
@@ -156,7 +135,7 @@ func TestSchemeHost(t *testing.T) {
 				},
 			},
 			expectedScheme: "http",
-			expectedHost:   "route-namespace.router.default.svc.cluster.local:80",
```

Comment:
can you make sure we have tests capturing all the headers we see for all four of those cases?